lib/sda/node.go

```diff
@@ -0,0 +1,270 @@
+package sda
+
+import (
+	"errors"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/crypto/abstract"
+	"github.com/satori/go.uuid"
+	"reflect"
+)
+
+/*
+Node represents a protocol-instance in a given TreeNode. It is linked to
+Overlay where all the tree-structures are stored.
+*/
+
+type Node struct {
+	overlay *Overlay
+	token   *Token
+	// channels holds all channels available for the different message-types
+	channels map[uuid.UUID]interface{}
+	// registered handler-functions for that protocol
+	handlers map[uuid.UUID]MsgHandler
+	// The protocolInstance belonging to that node
+	instance ProtocolInstance
+	// aggregate messages in order to dispatch them at once in the protocol
+	// instance
+	msgQueue map[uuid.UUID][]*SDAData
+	// Holds flags that influence the behaviour of the node
+	flags uint32
+}
+
+const (
+	NCAggregateMessages = 1 << iota
+	NCTimeout           = 1 << iota
```

Comment:
Yep, gives 1, 2, 4, 8, 16, ... Using iota only once is OK if you just want to linearly increase. Else you have to repeat the iota.

https://golang.org/ref/spec#Iota